### DISCUSSION TITLE
Quantity Value Conversions per Object

### DISCUSSION THREAD

**Discussion by cancan101** [pimcore/pimcore#17600] [Q&A / Support]
Is there any way to define the factor used for quantity conversion on a per object basis? For example I have units: `EA` - Eaches and `PL` - Pallets. How many `EA -> Pallets` varies based on the specific product. Is it possible to do that in Pimcore using the [Quantity Values](https://pimcore.com/docs/platform/Pimcore/Objects/Object_Classes/Data_Types/Number_Types/#quantity-value).

There is a concept of ["Dynamic unit conversion"](https://pimcore.com/docs/platform/Pimcore/Objects/Object_Classes/Data_Types/Number_Types/#dynamic-unit-conversion); however, the `convert` method:
https://github.com/pimcore/pimcore/blob/68892689cc17d299ead2dce18303db8b00da6f76/models/DataObject/QuantityValue/QuantityValueConverterInterface.php#L21-L24
does not get passed in the object itself.

**Comment by developkariyer** [comment]
In Product class I have numberInBox, numberInPallete, numberInContainer fields, which I set manually (sometimes required) or (if left blank) calculated onPreUpdate using given product dimensions.

In another object, which deals with shipment, I use a calculated field that checks all these values per product and give a final result.

What you asked is done in calculator class behind curtains in my example.
